Errors when manage categories

**Version Info**
master 18.1

**Describe the bug**
When adding/removing/editing categories, errors are displayed that are not necessarily related to the action being performed. The errors observed are:

Without console errors:
![image.png](https://images.zenhubusercontent.com/5e1c60ee5e668415ca9f76d9/7fff9ad6-fdf8-4d37-ad20-911a419ba702)

With console errors:
![image.png](https://images.zenhubusercontent.com/5e1c60ee5e668415ca9f76d9/68a2d42e-e336-4d6b-985e-4890b3714a72)
![image.png](https://images.zenhubusercontent.com/5e1c60ee5e668415ca9f76d9/4eee8f12-79c9-4e43-b27c-577ad78c29ca)

**How to Reproduce**
Steps to reproduce the behavior:
1. Go to oscm-portal/shop/manageCategories.jsf
2. ADD/REMOVE/EDIT CATEGORY
4. See error

**Observed behavior**
As above

**Expected behavior**
- [ ] During manage categories, errors should only appear for a specific reason
- [ ] Add/remove and edit commands should do the same
- [ ] Test test04addCategory() at PortalMarketplaceWT should pass everytime

**Additional context**
[manageCategories](https://app.zenhub.com/files/108258123/fb44231b-e744-491f-a3ca-1d92f0776b69/download)

Yes, it's something like I assumed. The version in one VOCategory object contained in the CategoryRow model is not set (version 0) and an update is attempted. The insert fails on the persistence side with the SQL exception shown in the attached system.log:

Caused by: java.sql.BatchUpdateException: Batch entry 0 insert into Category 
(categoryId, version, marketplaceKey, TKEY) _values ('Category 1', **0**, 11000, 10001)_ 
was aborted.  

This unexpected exception is not handled and therefore results in the "Operation failed" screen.

--- a/oscm-ui-tests/src/test/java/org/oscm/portal/PortalMarketplaceWT.java
+++ b/oscm-ui-tests/src/test/java/org/oscm/portal/PortalMarketplaceWT.java
@@ -11,7 +11,11 @@ package org.oscm.portal;
 
 import static org.junit.Assert.assertTrue;
 
-import org.junit.*;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Rule;
+import org.junit.Test;
 import org.junit.rules.TestWatcher;
 import org.junit.runners.MethodSorters;
 import org.openqa.selenium.By;
@@ -91,8 +95,6 @@ public class PortalMarketplaceWT {
     assertTrue(tester.getExecutionResult());
   }
 
-  // FIXME test ignored due #787
-  @Ignore
   @Test
   public void test04addCategory() throws Exception {
     tester.visitPortal(PortalPathSegments.MANAGE_CATEGORIES);
@@ -103,7 +105,8 @@ public class PortalMarketplaceWT {
         PortalHtmlElements.MANAGE_CATEGORIES_CATEGORY_ID_INPUT,
         "CategoryId" + PlaygroundSuiteTest.currentTimestampe);
     tester.writeValue(PortalHtmlElements.MANAGE_CATEGORIES_CATEGORY_NAME_INPUT, "CategoryName");
-    Thread.sleep(2000);
+    tester.waitForElement(By.id(PortalHtmlElements.MANAGE_CATEGORIES_ADD_BUTTON), 10);
+    Thread.sleep(1000);
     tester.clickElement(PortalHtmlElements.MANAGE_CATEGORIES_SAVE_BUTTON);
 
     assertTrue(tester.getExecutionResult());

